some of our tests now fail due to pytest=5.4.0

Some of Omar's data-regridding tests are now failing because they rely on an "accidental" feature which has now been "fixed". See https://github.com/pytest-dev/pytest/issues/6909 for more details (and it's not just us that has been bitten by this).

For reference, the tests fail with a message like

```
==================================== test session starts =====================================
platform linux -- Python 3.8.1, pytest-5.4.0, py-1.8.1, pluggy-0.13.1
rootdir: /home/dburke
collected 2329 items / 1 error / 2328 selected

=========================================== ERRORS ===========================================
_ ERROR collecting anaconda/envs/sherpa-wtf/lib/python3.8/site-packages/sherpa-4.12.0+121.g2d36b10e.dirty-py3.8-linux-x86_64.egg/sherpa/models/tests/test_regrid_unit.py _
In function "test_evaluation_space2d_overlaps":
Parameter "x_overlaps" should be declared explicitly via indirect or in function itself
!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!
====================================== 1 error in 1.23s ======================================
```

It also does not hit all our test configurations on Travis; I don't know if that's just that the new version hasn't made it to all set ups yet or something more interesting on how we set things up.

Explicitly setting the `indirect` argument, as shown below, makes the tests run, but they now fail

```python
@pytest.mark.parametrize('x_overlaps, y_overlaps', [
    (True, True),
    (True, False),
    (False, False),
], indirect=True)
@pytest.mark.parametrize('integrated', [
    True, False
], indirect=True)
def test_evaluation_space2d_overlaps(setup_overlapping_spaces):
```

which creates

```
=========================================== ERRORS ===========================================
_____________ ERROR at setup of test_evaluation_space2d_overlaps[True-True-True] _____________
file /home/dburke/anaconda/envs/sherpa-wtf/lib/python3.8/site-packages/sherpa-4.12.0+121.g2d36b10e.dirty-py3.8-linux-x86_64.egg/sherpa/models/tests/test_regrid_unit.py, line 852
  @pytest.mark.parametrize('x_overlaps, y_overlaps', [
      (True, True),
      (True, False),
      (False, False),
  ], indirect=True)
  @pytest.mark.parametrize('integrated', [
      True, False
  ], indirect=True)
  def test_evaluation_space2d_overlaps(setup_overlapping_spaces):
file /home/dburke/anaconda/envs/sherpa-wtf/lib/python3.8/site-packages/sherpa-4.12.0+121.g2d36b10e.dirty-py3.8-linux-x86_64.egg/sherpa/models/tests/test_regrid_unit.py, line 884
  @pytest.fixture
  def setup_overlapping_spaces(integrated, x_overlaps, y_overlaps):
E       fixture 'integrated' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, capture_all_warnings, clean_astro_ui, clean_ui, doctest_namespace, make_data_path, monkeypatch, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, restore_xspec_settings, run_thread, setup_1d, setup_overlapping_spaces, test_data_path, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/dburke/anaconda/envs/sherpa-wtf/lib/python3.8/site-packages/sherpa-4.12.0+121.g2d36b10e.dirty-py3.8-linux-x86_64.egg/sherpa/models/tests/test_regrid_unit.py:884
____________ ERROR at setup of test_evaluation_space2d_overlaps[True-True-False] _____________
...
```

Oooh. It looks like we just need to add `x_overlaps, y_overlaps, integrated` to the argument list to `test_evaluation_space2d_overlaps` and everything will be golden. Expect a PR.

--- a/sherpa/models/tests/test_regrid_unit.py
+++ b/sherpa/models/tests/test_regrid_unit.py
@@ -1,5 +1,5 @@
 #
-#  Copyright (C) 2017, 2018, 2019  Smithsonian Astrophysical Observatory
+#  Copyright (C) 2017, 2018, 2019, 2020  Smithsonian Astrophysical Observatory
 #
 #
 #  This program is free software; you can redistribute it and/or modify
@@ -439,7 +439,7 @@ def test_regrid1d_error_grid_mismatch_1(setup_1d):
 
 def test_ui_regrid1d_non_overlapping_not_allowed():
     """Integrated data space must not overlap"""
-    
+
     ui.dataspace1d(1,100,2,dstype=Data1DInt)
     b1 = Box1D()
     ui.set_model(b1)
@@ -448,7 +448,7 @@ def test_ui_regrid1d_non_overlapping_not_allowed():
     b1.ampl.max=100
     grid_hi = np.linspace(2,101,600)
     grid_lo = np.linspace(1,100,600)
-    with pytest.raises(ModelErr) as excinfo:    
+    with pytest.raises(ModelErr) as excinfo:
         rb1 = b1.regrid(grid_lo,grid_hi)
 
     assert ModelErr.dict['needsint'] in str(excinfo.value)
@@ -462,7 +462,7 @@ def test_low_level_regrid1d_non_overlapping_not_allowed():
     c = Box1D()
     lo = np.linspace(1,100,600)
     hi = np.linspace(2,101,600)
-    with pytest.raises(ModelErr) as excinfo:    
+    with pytest.raises(ModelErr) as excinfo:
         c.regrid(lo, hi)
 
     assert ModelErr.dict['needsint'] in str(excinfo.value)
@@ -856,7 +856,7 @@ def test_evaluation_space2d_start_end(xlo, xhi, ylo, yhi):
 @pytest.mark.parametrize('integrated', [
     True, False
 ])
-def test_evaluation_space2d_overlaps(setup_overlapping_spaces):
+def test_evaluation_space2d_overlaps(x_overlaps, y_overlaps, integrated, setup_overlapping_spaces):
     x1, y1, x2, y2, overlaps = setup_overlapping_spaces
 
     space_one = EvaluationSpace2D(x=x1[0], xhi=x1[1], y=y1[0], yhi=y1[1])

